modal-go/function_call.go

```diff
@@ -0,0 +1,44 @@
+package modal
+
+import (
+	"context"
+	"fmt"
+
+	pb "github.com/modal-labs/libmodal/modal-go/proto/modal_proto"
+)
+
+// FunctionCall references a Modal Function Call. Function Calls are
+// Function invocations with a given input. They can be consumed
+// asynchronously (see Get()) or cancelled (see Cancel()).
+type FunctionCall struct {
+	FunctionCallId string
+	ctx            context.Context
+}
+
+// Gets the ouptut for a FunctionCall
+func (fc *FunctionCall) Get() (any, error) {
+	return pollFunctionOutput(fc.ctx, &fc.FunctionCallId)
+}
+
+// Lookup a FunctionCall
+func FunctionCallLookup(ctx context.Context, functionCallId string) (*FunctionCall, error) {
```

Comment:
To match the Python SDK this should be called `FunctionCallFromId`. Also this doc comment should start with the name of the function and be a complete sentence.